### ISSUE TITLE
Bot fails to Close connection after failing to connect due to being banned

### ISSUE THREAD

**Issue by Srki94** [martindevans/MumbleSharp#48]
Hi there. First of all, thank you for creating Mumble Sharp. Without it, my application wouldn't be possible! 

I have an issue that I don't know how to handle due to a lack of experience with networking.

If Bot gets banned on a Server and tries to connect to it at any point in the future, I get 
`$exception	{"Unable to read data from the transport connection: An existing connection was forcibly closed by the remote host."}	System.IO.IOException
MumbleConnection.cs 293`

`_ssl.AuthenticateAsClient(serverName);` This line is where the exception is thrown.

If I try to handle this exception, I can't connect to any other server. Connection.Close fails on a new Connection attempt, due to _udp being null. 

``` 
public void Close()
        {
            State = ConnectionStates.Disconnecting;
            _udp.Close(); //  _udp is null at second connection attempt
            _tcp.Close();

            State = ConnectionStates.Disconnected;
        }
```

What can I do in this case and is there a way to let users know that the Bot was kicked/banned?

P.S. Creating a new instance of Client on every connection attempt seems to be working, but I'm not sure if it's a viable solution?

**Comment by Meetsch** [comment]
fixed in PR #47
You will be able to catch the exception on connection, and not have a NullReferenceException upon close.

**Comment by Meetsch** [comment]
Hi @Srki94,
Can you please report if our fix did effectively solve your issue. Thanks.
